pytorch_pfn_extras/training/extensions/log_report.py

```diff
@@ -29,12 +29,14 @@ def __call__(self, target, file_o):
             log = json.dumps(target, indent=4)
         elif self._format == 'json-lines':
             if self._append:
-                target = [target[-1]]
+                assert len(target) == 1
+                target = [target.pop()]
             # Add a new line at the end for subsequent appends
             log = '\n'.join([json.dumps(x) for x in target]) + '\n'
         elif self._format == 'yaml':
             if self._append:
-                target = [target[-1]]
+                assert len(target) == 1
+                target = [target.pop()]
```

Comment:
Could you modify `target` in `LogReport.__call__` instead? I think it's better for code readability by users.

Reply:
Fixed. PTAL.